tox.ini

```diff
@@ -2,10 +2,13 @@
 envlist = py{38,39,310,311,312}
 
 [testenv]
+allowlist_externals = bash
 deps =
+    pre-commit
     pytest
     pytest-cov
 commands =
+    bash -ec 'if [[ ${envpython} == *"py38"* ]]; then pre-commit run --all-files; fi' {posargs}
```

Comment:
I realized it does not make sense to pass the same posargs to here and pytest, right!?